src/Npgsql/NpgsqlConnection.cs

```diff
@@ -182,6 +183,34 @@ void GetPoolAndSettings()
             _pool = PoolManager.GetOrAdd(_connectionString, _pool);
         }
 
+        NpgsqlConnectionStringBuilder ParseConnectionString(string connectionString)
+        {
+            return Uri.TryCreate(connectionString, UriKind.Absolute, out var uri)
```

Comment:
We should probably also check that the scheme is [`postgres://` or `postgresql://`](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING).